src/mbgl/layermanager/fill_layer_factory.cpp

```diff
@@ -29,8 +29,13 @@ FillLayerFactory::createLayout(const LayoutParameters& parameters,
                                std::unique_ptr<GeometryTileLayer> layer,
                                const std::vector<Immutable<style::LayerProperties>>& group) noexcept {
     using namespace style;
-    using LayoutType = PatternLayout<FillBucket, FillLayerProperties, FillPattern>;
-    return std::make_unique<LayoutType>(parameters.bucketParameters, group, std::move(layer), parameters.imageDependencies);
+
+    using LayoutType = PatternLayout<FillBucket, FillLayerProperties, FillPattern, FillLayoutProperties>;
+    auto&& layout = std::make_unique<LayoutType>(parameters.bucketParameters, group, std::move(layer), parameters.imageDependencies);
+
+    layout->template maybeApplySortKey<FillSortKey>();
```

Comment:
why call it here? could it be done inside `Layout` constructor?

Reply:
I wanted to do this originally, but we don't have `constexpr if` so I can't have the conditional `SortKeyType` template param useable in the constructor without breaking type safety (`FillExtrusion` shares this constructor and and has no `SortKey` property.) I was thinking about making a `PatternLayoutSorted` subclass ...

Reply:
e.g.
```
template <typename SortKeyPropertyType>
float calculateSortKey(const PatternFeature& feature)  {
.....
}
template <>
float calculateSortKey<void>(const PatternFeature&)  {
 return 0.0f;
}
````

Reply:
This is a great solution thank you, I had thought about using `void` but didn't think to make a specialization on it. Thanks for helping me with this, very useful knowledge!